pulsar-common/src/main/java/org/apache/pulsar/common/util/FieldParser.java

```diff
@@ -212,7 +212,7 @@ public static <T> void setEmptyValue(String strValue, Field field, T obj)
             if (field.getType().equals(List.class)) {
                 field.set(obj, new ArrayList<>());
             } else if (field.getType().equals(Set.class)) {
-                field.set(obj, new HashSet<>());
+                field.set(obj, new LinkedHashSet<>());
```

Comment:
If order is required, it is recommended to use `List` instead of `Set`.

Reply:
It's true for new configuration fields. However, using `Set` ensures the uniqueness of the elements in the collection, so it's non-trivial to convert the type of these fields from `Set` to `List`, and there are quite a few existing configuration fields of the type `Set`. On the other hand, changing `HashSet` to `LinkedHashSet` makes all existing fields ordered and it seems harmless except for a bit increase in the per-element memory consumption which should be acceptable for configuration items.